src/diffusers/utils/hub_utils.py

```diff
@@ -129,3 +134,116 @@ def create_model_card(args, model_name):
 
     card_path = os.path.join(args.output_dir, "README.md")
     model_card.save(card_path)
+
+
+def extract_commit_hash(resolved_file: Optional[str], commit_hash: Optional[str] = None):
```

Comment:
all the added files are mostly copied from `transformers` with some small changes.